### DISCUSSION TITLE
How to always serve fresh content to editors so they see their changes immediately?

### DISCUSSION THREAD

**Discussion by maybebansky** [sanity-io/sanity#4948] [Sanity Studio Support]
For speed and cost savings I'd like to use the CDN version of the API. As the docs say, after Sanity content is edited it can take some time for this change to propagate: https://www.sanity.io/docs/api-cdn#d09568b4f4e3

> The entire cache for a dataset is invalidated whenever there is a write to a non-draft document in the dataset. Invalidations may take up to 60 seconds to fully propagate.

This is fine for normal users however if an editor makes a change and reloads the live website they will expect to see the change immediately. I imagine this is a common requirement so is there a reccomended solution?  

All I can think to do is make the user log in to the live site, and once authenticated use the non-CDN version of the API. However this isn't great for a number of reasons:

- There may be multiple clients consuming the API. Its annoying to have to log into all of them.
- I feel it makes me as a freelance look a little unprofessional explaining this limitation to clients, especially if they're used to Wordpress, Drupal etc. 

I'm new to headless CMSs so not sure if this is just a limitation that comes with the territory?

**Comment by stipsan** [comment]
Depending on your framework and setup what you want is to use our GROQ powered webhooks in combination with whatever feature your framework offers for ["On Demand Revalidation"](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#on-demand-revalidation), or "On Demand ISR/Incremental Static Regeneration".
